route/build/src/standalone/model_setup.f90

```diff
@@ -256,32 +295,102 @@ SUBROUTINE inFile_pop(ierr, message)  ! output
-    infileinfo_data(iFile)%iTimebound(2) = infileinfo_data(iFile-1)%iTimebound(2) + nTime ! the last index from the perivous nc file + 1
+    inputfileinfo(iFile)%iTimebound(1) = inputfileinfo(iFile-1)%iTimebound(2) + 1 ! the last index from the perivous nc file + 1
+    inputfileinfo(iFile)%iTimebound(2) = inputfileinfo(iFile-1)%iTimebound(2) + nTime ! the last index from the perivous nc file + 1
    endif
 
   enddo
 
   ! close ascii file
   close(unit=unit,iostat=ierr); if(ierr/=0)then;message=trim(message)//'problem closing forcing file list'; return; end if
 
-  ! passing the first nc file as global file name to read
-  fname_qsim = trim(infileinfo_data(1)%infilename)
-
  END SUBROUTINE inFile_pop
 
 
+ ! *********************************************************************
+ ! private subroutine: to synchronize the iTimebound of
+ ! the inputfileinfo_wm to match the inputfileinfo
+ ! *********************************************************************
+ SUBROUTINE inFile_sync_time(inputfileinfo,      & ! input: the structure of simulated runoff, evapo and
+                             inputfileinfo_wm,   & ! inout: input file information
+                             ierr, message)        ! output: error control
+
+  ! data types
+  USE dataTypes, ONLY: infileinfo               ! the data type for storing the infromation of the nc files and its attributes
+
+  ! public data
+  USE public_var, ONLY: time_units    ! time units (seconds, hours, or days)
+  USE public_var, ONLY: dt            ! simulation time step in seconds
+  USE public_var, ONLY: secprday      ! conversion of steps in days to seconds
+
+  ! input
+  type(infileinfo),         intent(in)              :: inputfileinfo(:)       ! the name of structure that hold the infile information
+
+  ! inout
+  type(infileinfo),         intent(inout)           :: inputfileinfo_wm(:)    ! the name of structure that hold the infile information
+
+  ! output
+  integer(i4b),             intent(out)             :: ierr             ! error code
+  character(*),             intent(out)             :: message          ! error message
+
+  ! local
+  integer(i4b)                                      :: nt
+  integer(i4b)                                      :: nFile             ! number of nc files for the simulated runoff
+  integer(i4b)                                      :: nFile_wm          ! number of nc files for the water managent
+  integer(i4b)                                      :: iFile             ! for loop over the nc files
+  real(dp)                                          :: refJulday_local   ! the reference julian day based on the first nc file
+  real(dp)                                          :: day_start_diff    ! conversion of the day to the local time
+  real(dp)                                          :: day_end_diff      ! conversion of the day to the local time
+
+
+  ! initialize error control
+  ierr=0; message='inFile_corr_time/'
```

Comment:
Fix message to match up with subroutine name

Reply:
fixed